library/std/src/fs/tests.rs

```diff
@@ -1953,3 +1953,24 @@ fn test_rename_directory_to_non_empty_directory() {
 
     error!(fs::rename(source_path, target_path), 145); // ERROR_DIR_NOT_EMPTY
 }
+
+#[test]
+fn test_rename_symlink() {
+    let tmpdir = tmpdir();
+    let original = tmpdir.join("original");
+    let dest = tmpdir.join("dest");
+
+    symlink_file("does not exist", &original).unwrap();
+    fs::rename(&original, &dest).unwrap();
+}
+
+#[test]
+#[cfg(windows)]
+fn test_rename_junction() {
+    let tmpdir = tmpdir();
+    let original = tmpdir.join("original");
+    let dest = tmpdir.join("dest");
+
+    junction_point("does not exist", &original).unwrap();
+    fs::rename(&original, &dest).unwrap();
+}
```

Comment:
Should these tests include a check that the symlink is accessible via `dest`?

The changes lgtm

Reply:
Oh wait, maybe I don't have to test it literally works just that `fs::read_link` returns that it's indeed a link to the file we expected.

Reply:
Ok, I did the above.